### DISCUSSION TITLE
ERR_PNPM_NO_MATCHING_VERSION_INSIDE_WORKSPACE  In : No matching version found for @cutting/util@* inside the workspace

### DISCUSSION THREAD

**Discussion by dagda1** [pnpm/pnpm#4561] [Q&A]
I am getting this error in a monorepo for a number of packages that have been published to npm.

For example if you run:

`pnpm add @cutting/hooks`

You get this error:

> ERR_PNPM_NO_MATCHING_VERSION_INSIDE_WORKSPACE  In : No matching version found for @cutting/util@* inside the workspace

The package [@cutting/hook package.json file](https://github.com/dagda1/cuttingedge/blob/main/packages/hooks/package.json#L26) uses the workspace protocol to reference [@cutting/util](https://github.com/dagda1/cuttingedge/blob/main/packages/util/package.json).

Should I remove the workspace protocol?

**Comment by zelal-dev** [comment]
@dagda1 How did you solve this? I have same issue

**Reply by ASR4** [reply]
@zelal-dev I have the same issue. Were you able to resolve this? What did you do?

**Comment by dagda1** [comment]
@zelal-dev unfortunately I cannot remember but fixed it is.  [This](https://github.com/pnpm/pnpm/issues/4348) might help

**Comment by maks-io** [comment]
In case anybody stumbles across this, for me my repository was locally corrupted and i was missing some `package.json` files in the individual packages in `packages/*`, therefore this kind of error message... so in that case it would be like a missing `package.json` file in `@cutting/util` for instance. Maybe it helps someone.

**Reply by julianpitt** [reply]
I was able to solve this issue when I migrating from npm  with nx to pnpm by running `pnpm -r update`
